packages/x-charts/src/internals/consumeSlots.tsx

```diff
@@ -105,16 +105,14 @@ export const consumeSlots = <
     }
 
     // Component can be wrapped in React.forwardRef or just a function that accepts (props, ref).
-    // If it is a plain function which accepts two arguments, we need to wrap it in React.forwardRef.
-    const OutComponent = (
-      Component.length >= 2 ? React.forwardRef(Component) : Component
-    ) as React.FunctionComponent<Props>;
+    // If it is a plain function which accepts two arguments, we don't need to pass a ref because the component doesn't expect it
+    const shouldPassRef = Component.length >= 2;
 
     if (process.env.NODE_ENV !== 'production') {
-      OutComponent.displayName = `${name}.slots.${slotPropName}`;
+      Component.displayName = `${name}.slots.${slotPropName}`;
```

Comment:
Also, it is a good idea to change the `displayName` of a user's component? Not sure if it's more confusing than helpful. I'm thinking of the use case of a user searching for their component in React DevTools and not being able to find it.

Reply:
The name change was mostly because of the ForwardRef which would wrap it into a new function, I just didn't check if a user component was passed.

Now we can probably remove it.

Reply:
Should we remove the rename?